tests/jax/integration/executor_test.py

```diff
@@ -13,265 +13,294 @@
-            extras_spec=ExtrasLogProbSpec,
-            parameter_server=DefaultParameterServer,
-            executor_parameter_client=ExecutorParameterClient,
-            **executor,
-            distributor=Distributor,
-            trainer_parameter_client=mocks.MockTrainerParameterClient,
-            logger=mocks.MockLogger,
-            trainer=mocks.MockTrainer,
-            trainer_dataset=mocks.MockTrainerDataset,
-        )
-        return components, {}
 
+def test_executor_multi_thread_with_adder(test_system_mt: System) -> None:
+    """Test if the executor instantiates processes as expected."""
 
-@pytest.fixture
-def test_system_except_trainer() -> System:
-    """Add description here."""
-    return TestSystemExceptTrainer()
-
-
-# Skip failing test for now
-@pytest.mark.skip
-def test_except_trainer(
-    test_system_except_trainer: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
-
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
+    (trainer_node,) = test_system_mt._builder.store.program._program._groups["trainer"]
+    (executor_node,) = test_system_mt._builder.store.program._program._groups[
+        "executor"
+    ]
+    (evaluator_node,) = test_system_mt._builder.store.program._program._groups[
+        "evaluator"
+    ]
+    (parameter_server_node,) = test_system_mt._builder.store.program._program._groups[
+        "parameter_server"
+    ]
+    trainer_node.disable_run()
+    executor_node.disable_run()
+    evaluator_node.disable_run()
+    parameter_server_node.disable_run()
+
+    test_system_mt.launch()
+    time.sleep(10)
+    executor = executor_node._construct_instance()
+
+    executor.run_episode()
+    # Observe first and observe
+    assert executor._executor.store.adder._writer.history
+    assert list(executor._executor.store.adder._writer.history.keys()) == [
+        "observations",
+        "start_of_episode",
+        "actions",
+        "rewards",
+        "discounts",
+        "extras",
+    ]
+    assert list(
+        executor._executor.store.adder._writer.history["observations"].keys()
+    ) == ["agent_0", "agent_1", "agent_2"]
+    assert (
+        type(executor._executor.store.adder._writer.history["observations"]["agent_0"])
+        == OLT
     )
 
-    # Networks.
-    network_factory = ippo.make_default_networks
-
-    # Build the system
-    test_system_except_trainer.build(
-        environment_factory=environment_factory,
-        network_factory=network_factory,
-        executor_parameter_update_period=20,
-        multi_process=False,
-        run_evaluator=True,
-        num_executors=1,
-        use_next_extras=False,
+    assert len(executor._executor.store.adder._writer._column_history) != 0
+
+    # Select actions and select action
+    i = 0
+    while (
```

Comment:
First of all yeah the launchpad now exit because of the reverb server stopping in both the trainer and executor test.
About the loop yeah we need it because when I run executor.run_episode it is a very slow thread that it causes that I test the action selection and observations before even getting updated

Reply:
Does executor.run_episode() not work synchronously? E.g. does the code not wait for the episode to complete first?

Reply:
No it won't wait for it to finish that's the problem :(

Reply:
Ah okay.